src/snowflake/cli/api/exceptions.py

```diff
@@ -188,3 +189,54 @@ def __init__(self, options: list[str]):
         super().__init__(
             f"Parameters {comma_separated_options} and {options_with_quotes[-1]} are incompatible and cannot be used simultaneously."
         )
+
+
+class WarehouseDoesNotExistOrRoleUnauthorizedError(ClickException):
```

Comment:
Why is warehouse special? We can probably just merge this with the more generic "DoesNotExist..." below.